src/node_buffer.cc

```diff
@@ -1223,6 +1223,34 @@ static void EncodeInto(const FunctionCallbackInfo<Value>& args) {
   results[1] = written;
 }
 
+static void IsUtf8(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  CHECK_EQ(args.Length(), 1);
+  CHECK(args[0]->IsTypedArray() || args[0]->IsArrayBuffer());
+
+  Local<ArrayBuffer> buf;
+  size_t offset = 0;
+  size_t length = 0;
+
+  if (args[0]->IsTypedArray()) {
+    Local<v8::TypedArray> input = args[0].As<v8::TypedArray>();
+    buf = input->Buffer();
+    offset = input->ByteOffset();
+    length = input->ByteLength();
+  } else {
+    buf = args[0].As<ArrayBuffer>();
+    length = buf->ByteLength();
+  }
+
+  if (buf->WasDetached()) {
+    return node::THROW_ERR_BUFFER_CONTEXT_NOT_AVAILABLE(isolate);
+  }
+
+  const char* external = static_cast<const char*>(buf->Data()) + offset;
+  args.GetReturnValue().Set(simdutf::validate_utf8(external, length));
```

Comment:
You don't have to manually unpack the argument, you can use `ArrayBufferViewContents`. The function can be reduced to this:
```c++
static void IsUtf8(const FunctionCallbackInfo<Value>& args) {
  Environment* env = Environment::GetCurrent(args);
  THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
  ArrayBufferViewContents<char> abv(args[0]);
  args.GetReturnValue().Set(simdutf::validate_utf8(abv.data(), abv.length()));
}
```
Sorry, I should've mentioned that in my previous review but I assumed you'd seen the common pattern in this file.

There's admittedly a difference: `ArrayBufferViewContents` doesn't check `WasDetached()`. Maybe it should.

Reply:
It seems that in order to check if `WasDetached(),` I have to add a function to `ArrayBufferViewContents`. Is there a difference in performance with the suggested change, or is this merely for aesthetics?

Reply:
I added `WasDetached()` to ArrayBufferViewContents

```
static void IsUtf8(const FunctionCallbackInfo<Value>& args) {
  Environment* env = Environment::GetCurrent(args);
  CHECK_EQ(args.Length(), 1);
  CHECK(args[0]->IsTypedArray() || args[0]->IsArrayBuffer());
  ArrayBufferViewContents<char> abv(args[0]);

  if (abv.WasDetached()) {
    return node::THROW_ERR_BUFFER_CONTEXT_NOT_AVAILABLE(env->isolate());
  }

  args.GetReturnValue().Set(simdutf::validate_utf8(abv.data(), abv.length()));
}
```

Reply:
I hope the new changes do not antagonize the C++ Gods :))

Reply:
> Is there a difference in performance with the suggested change

There is. `ArrayBufferViewContents` calls `CopyContents` for small arraybuffers because that's faster than going through `BackingStore`.

Reply:
Thanks! @bnoordhuis